app/main.py

```diff
@@ -1,8 +1,23 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    person_list = []
+    for person_data in people:
+        person = Person(person_data["name"], person_data["age"])
+        person_list.append(person)
+
+    for spouse in people:
+        person = Person.people[spouse["name"]]
+        if "wife" in spouse and spouse["wife"]:
+            person.wife = Person.people[spouse["wife"]]
+        elif "husband" in spouse and spouse["husband"]:
+            person.husband = Person.people[spouse["husband"]]
+
+    return person_list
```

Comment:
Code Quality Issue: This function does two things: it creates `Person` instances and assigns spouses. According to the Single Responsibility Principle, a function should do one thing. Consider splitting this function into two: one for creating `Person` instances and another for assigning spouses.